syntax = "proto3";

import "eigen.proto";

message SamplerParams {

	message NormalGammaParams {
		double mu0 = 1;
		double a = 2;
		double b = 3;
		double lam_ = 4;
	}

	message BetaParams {
		double a = 1;
		double b = 2;
	}

	message InverseWishartParams {
		double nu = 1;
		bool identity = 2;
		EigenMatrix V = 3;
	}

	message InverseGammaParams {
		double alpha = 1;
		double beta = 2;
	}

	message RhoParams {
	  oneof rhoparams_type {
	    double fixed = 1;
	    BetaParams beta_prior = 2;
	  };
	}

	message VarianceParams {
		oneof varianceparams_type {
			InverseWishartParams inv_wishart_prior = 1;
			InverseGammaParams inv_gamma_prior = 2;
		};
	}

	message GraphParams {
		oneof graphparams_type {
			double fixed = 1;
			BetaParams beta_prior = 2;
		};
	}

	int32 num_components = 1;
	NormalGammaParams p0_params = 2;
	RhoParams rho = 3;
	VarianceParams sigma = 4;
	double mtilde_sigmasq = 5;
	GraphParams graph_params = 6;
}

